algo.py

```diff
@@ -1,14 +1,15 @@
 from __future__ import division
-from sklearn.linear_model import LinearRegression, Ridge
-from sklearn.ensemble import RandomForestRegressor
-from sklearn.pipeline import Pipeline
+
 import re
-import pandas as pd
-from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer
+
+from pandas import DataFrame
```

Comment:
A rule of thumb that I've seen thrown about here and there and that makes sense to me is to try to stick to `import X` for standard library imports (familiarity, name-spacing) and to use `from X import Y` for third party code which tends to have lots of nested modules (e.g. SciKit) and where classes are generally speaking already name-spaced (e.g. DataFrame is very recognizably from Pandas).